### PR TITLE
chore: improve tsconfig strictness and ci workflows

### PR DESCRIPTION
**Pull request by james-elicx**
- improved the workflows for checks + e2es.
- stricter tsconfig rules.
- type declarations for process.env.
- better types for fetch handler.

### CODE REVIEW

File: packages/cloudflare/src/cli/templates/worker.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ export default {
           .default as typeof import("next/dist/server/next-server").default;
 
         requestHandler = new NextNodeServer({
-          conf: { ...nextConfig, env },
+          conf: nextConfig,
```
Comment (james-elicx) [inline packages/cloudflare/src/cli/templates/worker.ts:45]: note: nextconfig env option is for inlining variables during build. that isn't applicable for us here.

File: package.json [hunk 2]
```diff
@@ -8,10 +8,12 @@
     "prettier": "3.3.3"
   },
   "scripts": {
-    "prettier:check": "prettier --check .",
+    "prettier": "prettier --check .",
     "prettier:fix": "prettier --write .",
-    "lint:check": "pnpm -r lint:check",
+    "lint": "pnpm -r lint:check",
```
Comment (dario-piotrowicz) [inline package.json:13]: like the above, also note that you're introducing the inconsistency:
- to check linting at the root you run `lint`
- but in a package you run `lint:check`

File: packages/cloudflare/env.d.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+declare global {
+  namespace NodeJS {
+    interface ProcessEnv {
+      ASSETS: Fetcher;
+      __NEXT_PRIVATE_STANDALONE_CONFIG?: string;
+      SKIP_NEXT_APP_BUILD?: string;
+      [key: string]: string | Fetcher;
```
Comment (dario-piotrowicz) [inline packages/cloudflare/env.d.ts:7]:
```suggestion
      [key: string]: string;
```

is `Fetcher` necessary?

**Review by dario-piotrowicz — APPROVED** [review]
Looks good to me 🙂 

with a few minor comments

Thanks for the improvements! 🫶

File: packages/cloudflare/src/cli/build/patches/to-investigate/wrangler-deps.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import fs, { writeFileSync } from "node:fs";
+import { readFileSync, statSync, writeFileSync } from "node:fs";
```
[the remark this answers is not in the document]
Comment (james-elicx) [reply packages/cloudflare/src/cli/build/patches/to-investigate/wrangler-deps.ts:1]: fs. in this file was inconsistent with every other file in the codebase...

File: packages/cloudflare/env.d.ts (hunk 3 above)
Comment (james-elicx) [reply packages/cloudflare/env.d.ts:7]: not particularly

File: package.json [hunk 5]
```diff
@@ -12,7 +12,9 @@
     "prettier:fix": "prettier --write .",
     "lint:check": "pnpm -r lint:check",
     "lint:fix": "pnpm -r lint:fix",
-    "code:checks": "pnpm lint:check && pnpm lint:check",
+    "ts:check": "pnpm -r ts-check",
```
Comment (james-elicx) [inline package.json:15]:
```suggestion
    "ts:check": "pnpm -r ts:check",
```

File: packages/cloudflare/src/api/get-cloudflare-context.ts [hunk 6]
```diff
@@ -16,7 +16,7 @@ export type CloudflareContext<
   /**
    * the request's [cf properties](https://developers.cloudflare.com/workers/runtime-apis/request/#the-cf-property-requestinitcfproperties)
    */
-  cf: CfProperties;
+  cf: CfProperties | undefined;
```
[the remark this answers is not in the document]
Comment (dario-piotrowicz) [reply packages/cloudflare/src/api/get-cloudflare-context.ts:19]: oh.... interesting 😕

File: packages/cloudflare/src/cli/build/patches/to-investigate/wrangler-deps.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (dario-piotrowicz) [reply packages/cloudflare/src/cli/build/patches/to-investigate/wrangler-deps.ts:1]: oh 😮 ok 👍

File: .github/workflows/checks.yml [hunk 7]
```diff
@@ -1,19 +1,36 @@
...
 jobs:
-  test:
-    timeout-minutes: 60
+  checks:
+    name: ${{ matrix.script }}
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        script: [prettier:check, lint:check, ts:check, test]
```
Comment (james-elicx) [inline .github/workflows/checks.yml:16]:
```suggestion
        script: ["prettier:check", "lint:check", "ts:check", "test"]
```

File: packages/cloudflare/env.d.ts [hunk 8]
```diff
@@ -0,0 +1,12 @@
+declare global {
+  namespace NodeJS {
+    interface ProcessEnv {
+      ASSETS: Fetcher;
+      __NEXT_PRIVATE_STANDALONE_CONFIG?: string;
+      SKIP_NEXT_APP_BUILD?: string;
+      [key: string]: string;
```
Comment (james-elicx) [inline packages/cloudflare/env.d.ts:7]:
```suggestion
      [key: string]: string | Fetcher;
```

**Pull request merged**